benchalerts/benchalerts/pipeline_steps/github.py

```diff
@@ -31,11 +32,14 @@ class GitHubCheckStep(AlertPipelineStep):
         A GitHubRepoClient instance. Either provide this or ``repo``.
     commit_hash
         The commit hash to update. Default is to use the same one that was analyzed in
-        the GetConbenchZComparisonStep earlier in the pipeline.
+        the ``GetConbenchZComparisonStep`` earlier in the pipeline. If
+        ``GetConbenchZComparisonForRunsStep`` is used because the contender lacks commit
+        metadata, ``commit_hash`` must be provided.
```

Comment:
Pretty sure this is right; pls correct me if not.

Reply:
This defaulting behavior seems a little too magic now. I wonder if we should remove the default of this arg.

Reply:
That's fine, though we'll need to update all the implementations. I guess we need to anyway to specify `baseline_run_type` though

Reply:
Removed, which meant I had to move the param to first since everything else has defaults, and `None` doesn't mean anything here. In tests we were mostly setting these names anyway, so updating wasn't bad; hopefully implementations are similar. And hopefully we're not relying on positional args in usages anywhere.